File: my-folder/0002-add-two-numbers/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def addTwoNumbers(self, l1, l2):
        dummy = ListNode(0)
        current = dummy 
        carr=0
        while (l1!=None or l2!=None or carr):
            sum = 0
            if l1!=None:
                sum = sum+l1.val
                l1 = l1.next
            if l2!=None:
                sum = sum+l2.val
                l2 = l2.next
            sum += carr
            carr = sum//10

            new_node = ListNode(sum % 10)
            current.next = new_node
            current = current.next
        return dummy.next

File: my-folder/0002-add-two-numbers/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public void helper(ListNode l1, ListNode l2 , int carry , ListNode res){
        if(l1==null && l2 ==null && carry == 0) return ;

        int n1 = (l1!=null) ? l1.val : 0;
        int n2 = (l2!=null) ? l2.val : 0;
        int num = n1 + n2 + carry;
        int digit = num%10;
        carry = num/10;

        res.next = new ListNode(digit);

        helper( (l1 != null ? l1.next : null), (l2 != null ? l2.next : null) , carry , res.next);
    }


    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
        ListNode result = new ListNode(0);
        helper(l1 ,l2 , 0 , result);

        return result.next;
    }
}
